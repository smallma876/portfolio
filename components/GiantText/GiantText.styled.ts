import styled from "styled-components";

interface WrapperProps {
  color?: string;
}

const Wrapper = styled.h1<WrapperProps>`
  margin: 0;
  font-size: 2rem;
  color: ${(props) => props.color};
`;

export const GiantTextStyles = {
  Wrapper,
};
