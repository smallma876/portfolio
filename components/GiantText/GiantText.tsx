import React, { ReactNode } from "react";
import { GiantTextStyles } from "./GiantText.styled";

interface GiantTextProps {
  children: ReactNode;
  color?: string;
}

const GiantText = ({ children, ...rest }: GiantTextProps) => {
  return (
    <GiantTextStyles.Wrapper {...rest}>{children}</GiantTextStyles.Wrapper>
  );
};

export default GiantText;
