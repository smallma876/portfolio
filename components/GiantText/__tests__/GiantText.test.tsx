import { render, screen } from "@testing-library/react";
import GiantText from "../GiantText";

const renderComponent = async () => {
  const { container } = await render(<GiantText>Test</GiantText>);

  return container;
};

describe("GiantText", () => {
  it("should render match snapshot", () => {
    const container = renderComponent();
    expect(screen.getByText("Test")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
