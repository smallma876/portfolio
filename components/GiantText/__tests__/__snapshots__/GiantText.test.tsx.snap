// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiantText should render match snapshot 1`] = `Promise {}`;
